ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -14495,6 +14495,8 @@ private void validateStorageFormat(
                           + "'='" + fileFormat + "')");
         }
       }
+    } else if (partitionTransformSpecExists) {
+      throw new SemanticException("Partition transforms are only supported by Iceberg storage handler");
```

Comment:
Hopefully, we will have a generic message like "The default Hive storage handler doesn't support partition transforms" so that we can add a new one that supportsPartitionTransform.

Reply:
I found we already had a discussion and intentionally chosen this message. So, it sounds ok as it is.
https://github.com/apache/hive/pull/5166#discussion_r1540632711